traits/core.md

```diff
@@ -0,0 +1,87 @@
+# Core traits
+
+## Manual Scaler
+
+This trait definition allows operators to manually scale the number of replicas for workload types that allow scaling operations.
+
+### Applicable Workloads
+
+ - `core.oam.dev/v1alpha2.containerWorkload`
+
+### Schematic
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `replicaCount` | `integer` | Y | `0` | The target number of replicas to create for a given component instance. |
+
+### Spec
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: TraitDefinition
+metadata:
+  name: manualscalertrait.core.oam.dev
+spec:
+  appliesToWorkloads:
+    - core.oam.dev/v1alpha2.containerWorkload
+  definitionRef:
+    schemaName: manualscalertrait.core.oam.dev
+```
+
+OAM runtime should store the schema with a version, the schema SHOULD be able to validate. For example using JSON Schema like this:
+
```

Comment:
As a core trait, the schema should be part of the specification. Are we saying that it doesn't have to be defined using JSON Schema?

Reply:
The schematic is listed above. It seems that alone should suffice.

As for the Schema, do we have to force its format? It seems that we just gave two schemas format below.

Reply:
Would it be worth picking a particular type of schema definition for all examples/reference schemas in the spec? Given we describe many of our non-schema examples using Kubernetes-compatible YAML I suggest we define schema using the OpenAPIv3 YAML syntax used below.

Reply:
I'm not sure about this. In my view, we have to introduce K8s CRD if we want to finish the example of workflow for end user. Or we won't be able to tell, how we could get the schema, and how to find the version, and how the schema definition is really defined.